Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Physics;
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.Utilities.Solvers;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Utilities
+{
+    /// <summary>
+    /// Tap to place is a far interaction component used to place objects on a surface.
+    /// </summary>
+    public class TapToPlace : Solver, IMixedRealityPointerHandler
+    {
+        [Experimental]
+        [SerializeField]
+        [Tooltip("The game object that will be placed if it is selected")]
+        private GameObject gameObjectToPlace;
+
+        /// <summary>
+        /// The game object that will be placed if it is selected
+        /// </summary>
+        public GameObject GameObjectToPlace
+        {
+            get => gameObjectToPlace;
+            set
+            {
+                if (value != null && gameObjectToPlace != value)
+                {
+                    gameObjectToPlace = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Check if a collider is present on the GameObjectToPlace
+        /// </summary>
+        public bool ColliderPresent
```

Comment:
After making this protected or private (probably private honestly? - we can expose things as people need them but this is just a helper function that seems to be specific to some one off checks we have), having a comment here about this not being called in an Update loop would be good (i.e. because it GetComponent-s which should really be done only once (maybe start/awake or in editor/inspector only)

Reply:
Changed `ColliderPresent` to `protected internal IsColliderPresent` and added `[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.SDK.Experimental.Inspectors")]` to enable access to tap to place's custom inspector.